test/benchmarks/material/button-harness/BUILD.bazel

```diff
@@ -0,0 +1,82 @@
+load("@npm_angular_dev_infra_private//benchmark/component_benchmark:component_benchmark.bzl", "component_benchmark")
+load("//tools:defaults.bzl", "ng_test_library", "ng_web_test_suite", "ts_library")
+
+package(default_visibility = ["//visibility:public"])
+
+# These are two separate files that provide the same tidy interface for running a performance
+# benchmark.
+#
+# I am not sure why the testbed version of the test does not log console.time to the terminal,
+# although I assume it has something to do with the default karma config that is generated by
+# karma_web_test_suite (which is wrapped by ng_web_test_suite).
+#
+# I am equally lost as to why the performance api exists for the testbed tests but not for the
+# protractor tests.
+#
+# If you have any idea why these things happen, please let me know - @wagnermaciel.
+
+ts_library(
+    name = "constants",
+    srcs = [":constants.ts"],
+)
+
+ts_library(
+    name = "protractor-benchmark-utilities",
+    srcs = [":protractor-benchmark-utilities.ts"],
+    deps = [
+        ":constants",
+        "@npm//@angular/dev-infra-private",
+    ],
+)
+
+ts_library(
+    name = "testbed-benchmark-utilities",
+    srcs = [":testbed-benchmark-utilities.ts"],
+    deps = [":constants"],
+)
+
+# ProtractorHarnessEnvironment
+
+component_benchmark(
+    name = "benchmark",
```

Comment:
```suggestion
    name = "e2e_benchmark",
```

Reply:
Since there are two benchmarks for different environments, we should disambiguate the target names.